### ISSUE TITLE
Proposal: --rcfile and --macrofile options for overriding search paths

### ISSUE THREAD

**Issue by daxelrod-datto** [debbuild/debbuild#114]
Debbuild hardcodes paths for loading macro definitions and rc:

https://github.com/debbuild/debbuild/blob/master/debbuild#L239-L248
https://github.com/debbuild/debbuild/blob/master/debbuild#L211-L222

This makes it difficult to run tests which invoke `debbuild` without playing games with chroots or containers, because we don't want tests influenced by whatever is installed.

Proposal:

- A new `--rcfile` command line option that works like the one in rpmbuild. When passed, it completely overrides the default search path:
--rcfile FILELIST
    Each of the files in the colon separated FILELIST is read sequentially by rpm for configuration information. Only the first file in the list must exist, and tildes will be expanded to the value of $HOME. The default FILELIST is /usr/lib/rpm/rpmrc:/usr/lib/rpm/redhat/rpmrc:/etc/rpmrc:~/.rpmrc. 
- A new `--macrofile` command line option with the same semantics, but for configuring the path list for loading macro definitions. I could imagine more pushback from this one since it diverges from what's available in rpmbuild.

If this makes sense, I'll work on implementing it.

**Comment by Conan-Kudo** [comment]
This is definitely a good idea, though I'd also suggest making a PR to implement this for rpm as well.

Being able to do this would allow for custom paths at runtime to override the main ones, which would be very useful for doing macros testing with rpm too.

I'd suggest going ahead and implementing `--rcfile` since that semantic exists in `rpm(8)` already, but starting with a PR for `--macrofile` to rpm first before implementing here, since I'd like to minimize the delta of features between us. If rpm upstream doesn't want it, I'm still okay with the idea of implementing it here, but I need to think about the semantics carefully for this.

**Comment by daxelrod-datto** [comment]
Alternative idea: rpm and rpmbuild already support --root

```
--root DIRECTORY
    Use the file system tree rooted at DIRECTORY for all operations. Note that this means the database within DIRECTORY will be used for dependency checks and any scriptlet(s) (e.g. %post if installing, or %prep if building, a package) will be run after a chroot(2) to DIRECTORY.
```

We could instead implement --root, although I'd prefer doing so without having to call chroot(2) if we can avoid it; since tests should be able to be run as any user.

**Comment by Conan-Kudo** [comment]
I don't think we want to do that. The inner semantics of `debbuild` using fakeroot and other things would break with `--root`. :/

**Comment by Conan-Kudo** [comment]
I've filed an issue with RPM to implement `rpm --macrofile`: https://github.com/rpm-software-management/rpm/issues/1469

**Comment by Conan-Kudo** [comment]
@daxelrod-datto Apparently, `--macrofile` exists as `--macros`, per https://github.com/rpm-software-management/rpm/issues/1469#issuecomment-754572938.

So you're good to go on implementing this by matching what RPM has here.
